src/fping.c

```diff
@@ -1104,11 +1104,11 @@ int main(int argc, char** argv)
         exit(num_noaddress ? 2 : 1);
     }
 
-    if (src_addr_set && socket4 >= 0) {
+    if (socket4 && (src_addr_set || socktype4 == SOCK_DGRAM)) {
```

Comment:
Shouldn't this still be `socket4 >= 0`? 0 is theoretically a valid fd while -1 is not

Reply:
Yes, thanks for catching this.